src/schema.js

```diff
@@ -179,6 +180,35 @@ class Schema {
     });
   }
 
+  mixinNamedParamFacade(api, def) {
+    function namedParamFacade(base, paramDef, ...args) {
+      let params = args;
+      if (args && args.length === 1 && typeof args[0] === 'object') {
+        const defaults = paramDef.reduce((result, item) => {
```

Comment:
Instead of going through `paramDef` and creating a `defaults` object each time, perhaps you could include this logic into the `isNamedParameterObject` logic?

Alternatively (contrary what I said face-to-face), it might be better (speed-wise) to compile a function for each function _once_, instead of having to run this logic each time someone calls a QIX method.